### PR TITLE
migrate oidc-discovery-provider to spire module

### PR DESCRIPTION
**Pull request by imjasonh**
NB: This actually renames `cgr.dev/chainguard/oidc-discovery-provider` to `cgr.dev/chainguard/spire-oidc-discovery-provider`, which I think is more correct anyway (it's the name of the package, for example).

If this sounds okay, we can see if anybody uses it that needs to be notified of the change.

### CODE REVIEW

File: images/spire/configs/latest.agent.apko.yaml [hunk 1]
```diff
@@ -22,4 +22,4 @@ entrypoint:
 annotations:
   "org.opencontainers.image.authors": "Chainguard Team https://www.chainguard.dev/"
   "org.opencontainers.image.url": https://edu.chainguard.dev/chainguard/chainguard-images/reference/spire-agent/
-  "org.opencontainers.image.source": https://github.com/chainguard-images/images/tree/main/images/spire-agent
+  "org.opencontainers.image.source": https://github.com/chainguard-images/images/tree/main/images/spire
```
Comment (mattmoor) [inline images/spire/configs/latest.agent.apko.yaml:25]: JFYI, we've been screwing this up everywhere, and I have a yellow sticky to audit these.

File: images/spire/main.tf [hunk 2]
```diff
@@ -26,7 +26,7 @@ module "latest-dev" {
   for_each = local.components
   source   = "../../tflib/publisher"
 
-  target_repository = var.target_repository
+  target_repository = "${var.target_repository}-${each.key}"
```
Comment (mattmoor) [inline images/spire/main.tf:29]: Thanks for fixing this, it was wrong before, but I think it's still wrong for the new component?

File: images/spire/tests/main.tf [hunk 3]
```diff
@@ -46,6 +47,15 @@ resource "helm_release" "spire" {
           tag        = data.oci_string.ref["agent"].pseudo_tag
         }
       }
+      spiffe-oidc-discovery-provider = {
+        enabled = true
+        config  = { acme = { tosAccepted = true } }
+        image = {
+          registry   = data.oci_string.ref["server"].registry
+          repository = data.oci_string.ref["server"].repo
+          tag        = data.oci_string.ref["server"].pseudo_tag
```
Comment (mattmoor) [inline images/spire/tests/main.tf:56]: wrong image

Comment (mattmoor) [inline images/spire/tests/main.tf:56]:
```suggestion
          registry   = data.oci_string.ref["oidc-discovery-provider"].registry
          repository = data.oci_string.ref["oidc-discovery-provider"].repo
          tag        = data.oci_string.ref["oidc-discovery-provider"].pseudo_tag
```

**Comment by mattmoor** [comment]
This LGTM.  Beware I did push several fixes to your fork, so fetch before you push anything.

We should sort out what sort of traffic this gets and plan a withdraw of the old tags as part of the migration to the new repo.

**Comment by imjasonh** [comment]
Looking through traffic I've only been able to identify our own traffic pushing and pulling the image to test it. Otherwise there are no logged-in users pulling the image, and even anonymous pulls look like our own testing in terms of useragents, timing and volume relative to our pushes.

I believe this change should be safe to make.

File: images/spire/configs/latest.agent.apko.yaml (hunk 1 above)
Comment (imjasonh) [reply images/spire/configs/latest.agent.apko.yaml:25]: Even better, we should be generating these so we can't mess them up.

**Review by mattmoor — APPROVED** [review]

**Pull request merged**